### PR TITLE
remove List and TextMap support from the JSON query language

### LINKED ISSUE #4855: Remove list and map matching from the JSON query language
The current list and map rules in the query language do not appear to be always what you would want; they would collide syntactically with alternative query language features for lists and maps, so by removing the ability to match them now, we make room for potentially different match support in the future.

From discussion with @leo-da @hurryabit @gerolf-da .
Related to (triggered by): #4779

### PR DESCRIPTION
**Pull request by S11001001**
Fixes #4855.

```rst
CHANGELOG_BEGIN
- [JSON API - Experimental] Removed support for maps and lists from the query language.
  See `issue #4855 <https://github.com/digital-asset/daml/issues/4855>`_.
CHANGELOG_END
```

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: docs/source/json-api/search-query-language.rst [hunk 1]
```diff
@@ -16,7 +16,7 @@ Fallback rule
 
 Unless otherwise required by one of the other rules below or to follow,
 values are interpreted according to :doc:`lf-value-specification`, and
-compared for equality.
+compared for equality, unless they are lists or maps.
```
Comment (hurryabit) [inline docs/source/json-api/search-query-language.rst]: Can we please make it more clear that lists and maps are not allowed in queries.

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/query/ValuePredicate.scala [hunk 2]
```diff
@@ -369,11 +316,8 @@ object ValuePredicate {
...
         }
-        case TextMap => {
-          case JsObject(q) =>
-            val elemTy = soleTypeArg("Map")
-            MapMatch(SortedLookupList(q) mapValue (fromValue(_, elemTy)))
-        }
+        case List | TextMap =>
+          predicateParseError(s"${typ.typ} not supported")
         case GenMap =>
           // FIXME https://github.com/digital-asset/daml/issues/2256
```
Comment (leo-da) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/query/ValuePredicate.scala]: the `FIXME` is not valid anymore. Does not make much sense to support `GenMap` queries when we are dropping `TextMap` query support.

File: docs/source/json-api/search-query-language.rst (hunk 1 above)
Comment (S11001001) [reply docs/source/json-api/search-query-language.rst]: How about c63f0917ef ?

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/query/ValuePredicate.scala (hunk 2 above)
Comment (S11001001) [reply ledger-service/http-json/src/main/scala/com/digitalasset/http/query/ValuePredicate.scala]: Combined in 4061ffc

**Comment by S11001001** [comment]
Will follow-up in a separate PR if further documentation updates are desired.

**Pull request merged**